### PR TITLE
Added legalMoves for King and Rook + some tests ( will add more tests )

### PR DESCRIPTION
**Pull request by cynzy**

### CODE REVIEW

File: src/Chess/ChessPieces.hs [hunk 1]
```diff
@@ -164,6 +207,77 @@ rowsBelow :: ChessPosition -> [Integer]
 rowsBelow (_, row) = if row == 1 then [] else (init [1..row])
 
 
+-- Get positions preceding (and including) one with enemy piece, or preceding (but not including) one with friendly piece (whichever is earlier)
```
Comment (azizsonawalla) [inline src/Chess/ChessPieces.hs]: I think these additions are duplicated

**Review by azizsonawalla — CHANGES_REQUESTED** [review]
Comments above

File: src/Chess/ChessPieces.hs (hunk 1 above)
Comment (cynzy) [reply src/Chess/ChessPieces.hs]: removed!

**Review by cynzy — COMMENTED** [review]
removed duplicate code

File: test/TestChessBoards.hs [hunk 2]
```diff
@@ -109,6 +113,20 @@ board11_H6 = [('G', 5), ('G', 7), ('F', 8)]
...
+board12_E5 = [('E', 6), ('D', 5), ('F', 5), ('F', 6), ('D', 4), ('F', 4)]
+
+
+-- https://lichess.org/editor/k3B3/2Q5/2r1pP2/P2pP1p1/Pb6/1P1PN2p/1R5P/K7_w_-_-_0_1
+board13 = fenToChessBoard "k3B3/2Q5/2r1pP2/P2pP1p1/Pb6/1P1PN2p/1R5P/K7"
+board13_A1 = [('A', 2), ('B', 1)] -- white king
+board13_A8 = [('A', 7), ('B', 8), ('B', 7)] -- black king
+board13_B2 = [('B', 1), ('A', 2), ('C', 2), ('D', 2), ('E', 2), ('F', 2), ('G', 2)] -- white rook
+board13_C6 = [('C', 7), ('C', 5), ('C', 4), ('C', 3), ('C', 2), ('C', 1), ('B', 6), ('A', 6), ('D', 6)] -- black rook
+board11_F8 = [('E', 7), ('G', 7), ('H', 6)]
```
Comment (azizsonawalla) [inline test/TestChessBoards.hs]:
```suggestion
board13_F8 = [('E', 7), ('G', 7), ('H', 6)]
```

**Review by azizsonawalla — CHANGES_REQUESTED** [review]
Other than the one small change, looks good! Thanks so much!

**Review by azizsonawalla — APPROVED** [review]

**Pull request merged**